### ISSUE TITLE
Could not create retry file '*.retry'. [Errno 2] No such file or directory: ''

### ISSUE THREAD

**Issue by kustodian** [ansible/ansible#16365]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 2.1.1.0 (stable-2.1 4d4bbcbb33) last updated 2016/06/20 08:55:21 (GMT +200)
```
##### CONFIGURATION

N/A
##### OS / ENVIRONMENT

Ubuntu
##### SUMMARY

If `retry_files_save_path` isn't set in ansible.cfg, when a playbook fails, a retry files tries to be created in an empty directory.
##### STEPS TO REPRODUCE

Make any playbook fail.
##### EXPECTED RESULTS

A try file should be created in the user home directory.
##### ACTUAL RESULTS

A warning message is printed and a retry file is not created:

```
 [WARNING]: Could not create retry file 'test_fail.retry'.         [Errno 2] No such file or directory: ''
```

It looks like the default value for the retry files isn't set to be `~/` like it's mentioned in the documentation.

**Comment by jimi-c** [comment]
The default value was changed in 2.0 (when we re-implemented retry files) to be the playbook dir when the value in the config was None (the new default).

**Comment by kustodian** [comment]
Shouldn't the default then be set to `./`, since if it's None, it throws an error like this. This should be easy to fix in the `constants.py`, or if you want to leave it as None, than it would probably best that Ansible does nothing, so it doesn't throw an error with default values like it currently does.

Since this was removed in 2.0, I think it would be best to remove `retry_files_enabled` options and just use `retry_files_save_path` which by default would be set to None, which would mean do not create retry files, so if you want to use retry files, you would set this one variable to the path where you want them to be created.

**Comment by vinamelody** [comment]
I'm also having this error on latest version 2.1.1.0 but it was ok using version 2.0.0.2 what's going on?

**Comment by sst1xx** [comment]
the same error for me
cat ansible/VERSION
2.1.1.0 1

**Comment by Pablohn26** [comment]
Same problem here:

> ansible --version
> ansible 2.1.1.0
>  config file = /etc/ansible/ansible.cfg
>  configured module search path = Default w/o overrides

On Fedora

> cat /etc/redhat-release 
> Fedora release 24 (Twenty Four)

**Comment by davidlt** [comment]
Hit he same on Fedora 23:

```
ansible 2.1.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

Fedora release 23 (Twenty Three)
```

**Comment by wernerflamme** [comment]
Same on my openSUSE Leap 42.1 and SLES 11 SP4, both running ansible 2.1.1.0

Both parameters were commented, i disabled the comments now.
`retry_files_enabled = False`
`retry_files_save_path = ~/.ansible-retry`
If retry_files_enabled was False by default, the `save_path` shouldn't matter anyway. But it does, the error is printed unless `enabled` is set to `False` explicitly. So both parameters seem to be neglected in the default settings.

**Comment by dagwieers** [comment]
Is this related to #17456 and fixed by #17457 ?

**Comment by Shaps** [comment]
@dagwieers yes it is the same as #17456 and should be fixed in #17457

**Comment by dagwieers** [comment]
resolved_by_pr #17457

**Comment by bcoca** [comment]
Fixed by #17457

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:

https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
